lib/compress/index.js

```diff
@@ -3611,9 +3612,12 @@ class Compressor extends TreeWalker {
                             if (!in_use || fixed_ids.has(def.id) && fixed_ids.get(def.id) !== node) {
                                 return maintain_this_binding(parent, node, node.right.transform(tt));
                             }
-                        } else if (!in_use) return make_node(AST_Number, node, {
-                            value: 0
-                        });
+                        } else if (!in_use) {
+                            // TODO: This produces "extra" zero number in var_catch_toplevel test
```

Comment:
@fabiosantoscode please take a look at this.

Reply:
> `undefined_variable++` is `NaN`, why is there even a zero there?

That's answer for your question :)

Reply:
Shouldn't it put a `NaN` literal there instead of a `0`?

Reply:
As I understand if variable not in use then it just replace it with zero

Reply:
I'd say let's put a `NaN` there just in case there's a bug somewhere else.

It's also more illustrative of the real output of the `unused_variable++` expression, and just in case people are calling `eval` on the code and it just so happens to be the last statement to be evaluated, Terser will output a `NaN` correctly. If a further pass doesn't remove it.

Reply:
You can remove the TODO comment too, since we know what this is doing. Or replace it with `// unused_variable++ === null` for future reference. Up to you.